src/vm/core.zig

```diff
@@ -213,6 +213,41 @@ pub const CairoVM = struct {
         _ = op_o;
     }
 
+    /// Attempts to deduce `op0` and `res` for an instruction, given `dst` and `op0`.
+    ///
+    /// # Arguments
+    /// - `inst`: The instruction to deduce `op0` and `res` for.
+    /// - `dst`: The destination of the instruction.
+    /// - `op0`: The first operand of the instruction.
+    ///
+    /// # Returns
+    /// - `Tuple`: A tuple containing the deduced `op0` and `res`.
+    pub fn deduceOp0(
+        self: *Self,
+        inst: *const instructions.Instruction,
+        dst: ?*const relocatable.MaybeRelocatable,
+        op1: ?*const relocatable.MaybeRelocatable,
+    ) !std.meta.Tuple(&[_]type{ ?relocatable.MaybeRelocatable, ?relocatable.MaybeRelocatable }) {
+        if (inst.opcode == .Call) {
```

Comment:
Maybe here we can use a match as we are dealing with an enum (`Opcode`) in order not to forget any variant in the logic.

Reply:
with switch i would need to do something like:
```zig
        switch (inst.opcode) {
            .Call => {
                var result: ?MaybeRelocatable = relocatable.newFromRelocatable(try self.run_context.pc.addUint(inst.size()));
                return .{ result, null };
            },
            .AssertEq => {
                _ = dst orelse return .{ null, null };
                _ = op1 orelse return .{ null, null };
                if ((inst.res_logic == .Add)) {
                    return .{ try subOperands(dst.?.*, op1.?.*), dst.?.* };
                } else if (dst.?.isFelt() and op1.?.isFelt() and
                    !op1.?.felt.isZero())
                {
                    return .{
                        relocatable.fromFelt(try dst.?.felt.div(op1.?.felt)),
                        dst.?.*,
                    };
                }
            },
            .NOp => {
                return .{ null, null };
            },
            .Ret => {
                return .{ null, null };
            },
        }
        return .{ null, null };

```

Cannot use `_` (prong) for last 2 case because it can only be used on non exhaustive enums.

Reply:
@lambda-0x A `else` would work no?

Reply:
oh right, rust habits of using `_` in match statements :)